src/components/Explore/TokenTable.tsx

```diff
@@ -53,20 +79,30 @@ export default function TokenTable() {
       </GridContainer>
     )
   } else if (error || data === null) {
-    return <GridContainer>Error Loading Top Token Data</GridContainer>
+    return (
+      <GridContainer>
+        <HeaderRow timeframe={timePeriod} />
+        <NoTokenDisplay>
+          <AlertTriangle size={16} />
+          An error occured loading tokens. Please try again.
+        </NoTokenDisplay>
+      </GridContainer>
+    )
   }
   /* if no favorites tokens */
   if (showFavorites && favoriteTokens.length === 0) {
     return (
       <GridContainer>
         <HeaderRow timeframe={timePeriod} />
-        <NoTokenDisplay>No Favorited Tokens</NoTokenDisplay>
+        <NoTokenDisplay>You have no favorited tokens</NoTokenDisplay>
       </GridContainer>
     )
   }
 
   const topTokenAddresses = Object.keys(data)
   const showTokens = showFavorites ? favoriteTokens : topTokenAddresses
+  const filteredTokens = useFilteredTokens(showTokens)
```

Comment:
@vm conditionally called here because we want to exit earlier if the data is null

Reply:
you need to put this at the top before you early exit, and have that hook return `[]`

Reply:
you also shouldn't pass `showTokens` in, call that hook inside `useFilteredTokens`

Reply:
ay yai captain